src/evaluators/ForOfStatement.js

```diff
@@ -254,21 +256,28 @@ export function ForInOfBodyEvaluation(realm: Realm, env: LexicalEnvironment, lhs
       } else { // g. Else,
         // i. If lhsKind is assignment, then
         if (lhsKind === "assignment") {
+          // This should be true given the algorithm. We add a check here mostly
+          // for Flow.
```

Comment:
No, I love invariants. I'm just surprised, somewhat dubious and very curious that this one helps Flow.